Code/src/main/java/nl/utwente/group10/ui/components/ConnectionAnchor.java

```diff
@@ -34,8 +38,45 @@ public ConnectionAnchor(Block block, CustomUIPane pane) throws IOException {
         fxmlLoader.setController(this);			
 
         fxmlLoader.load();
+        
+        setConnection(null);
     }
 
+    public void setConnection(Connection connection){
+    	this.connection = Optional.ofNullable(connection);
```

Comment:
Je kunt ook `Optional<Connection>` eisen (en een `Optional.empty` meegeven in de constructor) maar dit is ook wel prima.